src/mlpack/methods/ann/layer/noisylinear_impl.hpp

```diff
@@ -48,6 +48,50 @@ NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
   biasEpsilon.set_size(outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>::NoisyLinear(
+    NoisyLinear&& layer) :
+    inSize(std::move(layer.inSize)),
+    outSize(std::move(layer.outSize)),
+    weights(std::move(layer.weights))
+{
+  layer.inSize = 0;
+  layer.outSize = 0;
+  layer.weights = nullptr;
+  Reset();
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear const& layer)
+{
+  if( this != &layer)
+  {
+    inSize = layer.inSize;
+    outSize = layer.outSize;
+    weights = layer.weights;
+    Reset();
+  }
+  return *this;
+}
+
+template<typename InputDataType, typename OutputDataType>
+NoisyLinear<InputDataType, OutputDataType>&
+NoisyLinear<InputDataType, OutputDataType>::operator=(NoisyLinear&& layer)
+{
+  if (this != &layer)
+  {
+    inSize = std::move(layer.inSize);
+    layer.inSize = 0;
+    outSize = std::move(layer.outSize);
```

Comment:
Since `inSize` and `outSize` are primitive types, `std::move()` isn't necessary here.

Reply:
I see, I will make a change to that!